### PR TITLE
A few random fixes

### PR DESCRIPTION
**Pull request by naps62**
- Changes `buy()` to receive an amount in CTND instead of aUSD, making the whole API, and tests, a bit clearer
- Fixes withdraw function to only transfer funds not meant for refunds
- Fixes refundAmount and allocatedAmount to be 0 if an invalid cap is set

### CODE REVIEW

File: packages/contracts/src/deployConfigs.ts [hunk 1]
```diff
@@ -54,21 +55,22 @@ async function networkConfigs(chainId: number): Promise<Config> {
...
+    //   },
+    //   ctndSale2: {
+    //     start: 0, // TODO
+    //     end: 0, // TODO
+    //     supply: parseUnits("10000000"),
+    //   },
+    //   ctndVesting: {
+    //     start: 0, //TODO
+    //   },
+    // };
```
Comment (DavideSilva) [inline packages/contracts/src/deployConfigs.ts:73]: Is this commented code needed later on?

File: packages/contracts/contracts/token/Vesting.sol [hunk 2]
```diff
@@ -14,8 +14,6 @@ import {DateTime} from "../libraries/DateTime.sol";
 import "hardhat/console.sol";
```
Comment (DavideSilva) [inline packages/contracts/contracts/token/Vesting.sol:14]: We can probably remove this import, right? Or do we need to print something in this contract?

File: packages/contracts/contracts/token/Sale.sol [hunk 3]
```diff
@@ -10,6 +10,7 @@ import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.
 import {ISale} from "./ISale.sol";
 import {RisingTide} from "../RisingTide/RisingTide.sol";
 import {FractalRegistry} from "../fractal_registry/FractalRegistry.sol";
+import {Math} from "../libraries/Math.sol";
 
 import "hardhat/console.sol";
```
Comment (DavideSilva) [inline packages/contracts/contracts/token/Sale.sol:15]: We can probably remove this import, right? Or do we need to print something in this contract?

File: packages/contracts/src/deployConfigs.ts (hunk 1 above)
Comment (naps62) [reply packages/contracts/src/deployConfigs.ts:73]: It's commented while we don't have the final parameters for the mainnet deploy

File: packages/contracts/contracts/token/Vesting.sol (hunk 2 above)
Comment (naps62) [reply packages/contracts/contracts/token/Vesting.sol:14]: True

**Review by DavideSilva — APPROVED** [review]
:shipit:

**Pull request merged**